output_template/python/lmnet/utils/pool_interface.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import sys
+from multiprocessing import Pool
+
+
+class PoolInterface:
```

Comment:
About `PoolInterface`, I think `from multiprocessing import Pool` of python3 have not breaking change from python2.
The difference of python2, 3 is queue or Queue, Maybe.
https://python-future.org/compatible_idioms.html#queue

Do you have other reason of wrap multiprocessing in python2.7 and python3.5 ? 🤔 I think it is no need.

Reply:
@ruimashita 

Thanks for the review. Yeah, it has no breaking change between Python versions.
However, When I try to use it with OpenCV in Python3 It doesn't work and throws this error:
```
tensorflow/stream_executor/cuda/cuda_driver.cc:967] failed to alloc 2304 bytes on host: CUDA_ERROR_NOT_INITIALIZED
```


There is also a related issue here: https://github.com/opencv/opencv/issues/5150

So when I investigated this problem, some people suggested to use 
`from concurrent.futures import ThreadPoolExecutor` instead. Then it worked just fine. 💭 
Any thoughts?

Reply:
@Joeper214 
Oh, Thanks.
Maybe I think the reason of the error is the sub process wasn't initialized tensorflow. And when we use `thread` we will not get the error.

I come up with 3 ways.
1. Create sub process before initialize tensorflow.
Do `pool = Pool()` before `ProtobufLoader()` init. But It has a lot of changes, and I'm not sure it will work.

2. Using sub process, Initialize tensorflow in every `ProtobufLoader.run()`.
Add `import tensorflow as tf` in `run()`. It takes cost  in every `run()`....

3. Give up sub process, use `multiprocessing.dummy.Pool()` 
It is thread has the same interface as `multiprocessing.Pool`. Maybe work.

Reply:
`Pool` have `initializer` parameter, we maybe can set `initializer=nn.init`.
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool